crates/primitives/starknet/src/crypto/hash/mod.rs

```diff
@@ -20,19 +22,33 @@ impl ThreadSafeCopy for Hasher {}
 
 /// Implement the `HasherT` trait for the `Hasher` enum.
 impl HasherT for Hasher {
-    fn hash(&self, data: &[u8]) -> Felt252Wrapper {
+    fn hash_bytes(&self, data: &[u8]) -> Felt252Wrapper {
         match self {
-            Self::Pedersen(p) => p.hash(data),
-            Self::Poseidon(p) => p.hash(data),
+            Self::Pedersen(p) => p.hash_bytes(data),
+            Self::Poseidon(p) => p.hash_bytes(data),
         }
     }
 
-    fn hash_elements(&self, data: &[Felt252Wrapper]) -> Felt252Wrapper {
+    fn compute_hash_on_wrappers(&self, data: &[Felt252Wrapper]) -> Felt252Wrapper {
         match self {
-            Self::Pedersen(p) => p.hash_elements(data),
-            Self::Poseidon(p) => p.hash_elements(data),
+            Self::Pedersen(p) => p.compute_hash_on_wrappers(data),
+            Self::Poseidon(p) => p.compute_hash_on_wrappers(data),
         }
     }
+
+    fn hash_elements(_a: FieldElement, _b: FieldElement) -> FieldElement {
+        todo!()
```

Comment:
Done, but this comes with some downsides I think?

To be able to match `self` (otherwise I don't think there is a way to distinguish which variant of the enum it is), need to change the parameters of `hash_elements` and `compute_hash_on_elements` to accept `&self`.

Now, in quite a few places we actually used `<T as HasherT>::compute_hash_on_elements` but now that we need `self`, it's changed to `<T>::default().compute_hash_on_elements()`. Now, this is where it could become a DX small bump because while it's fine if we use `PedersenHasher` or `PoseidonHasher`, we need to make a choice for `Hasher` (which also implements `HasherT`) because it's ambiguous and its default is `Hasher::Pedersen(pedersen::PedersenHasher::default())`.

Reply:
If we really want to KISS can't we just not implement the HasherT trait on Hasher and just on Pedersen and Poseidon ? Would be cleaner no?

Reply:
As you mentionned the current state is way too ambiguous and the use of `default` should be prohibited

Reply:
`Default` impl. for `Hasher` has been made `unreachable` to preserve that feature for `Pedersen` and `Poseidon`.